soda/core/soda/execution/data_source.py

```diff
@@ -177,51 +183,104 @@ def sql_get_column(self, include_tables: list[str] | None = None, exclude_tables
 
-    def sql_find_table_names_includes_excludes(
-        self, include_tables: list[str] | None = None, exclude_tables: list[str] | None = None
+    def sql_find_table_names(
+        self,
+        filter: str | None = None,
+        include_tables: list[str] = [],
+        exclude_tables: list[str] = [],
+        table_column_name: str = "table_name",
+        schema_column_name: str = "table_schema",
     ) -> str:
-        table_filter_expression = self.sql_table_filter_based_on_includes_excludes(
-            "table_name", "table_schema", include_tables, exclude_tables
-        )
-        where_clause = f"\nWHERE {table_filter_expression} \n" if table_filter_expression else ""
-        return f"SELECT table_name \n" f"FROM {self.sql_information_schema_identifier()}" f"{where_clause}"
-
-    def sql_find_table_names(self, filter: str | None = None) -> str:
         sql = f"SELECT table_name \n" f"FROM {self.sql_information_schema_identifier()}"
         where_clauses = []
-        if self.schema:
-            where_clauses.append(f"lower(table_schema) = '{self.schema.lower()}'")
+
         if filter:
-            where_clauses.append(f"lower(table_name) like '{filter.lower()}'")
+            where_clauses.append(f"lower({table_column_name}) like '{filter.lower()}'")
+
+        includes_excludes_filter = self.sql_table_filter_based_on_includes_excludes(
+            table_column_name, schema_column_name, include_tables, exclude_tables
+        )
+        if includes_excludes_filter:
+            where_clauses.append(includes_excludes_filter)
+
         if where_clauses:
             where_clauses_sql = "\n  AND ".join(where_clauses)
             sql += f"\nWHERE {where_clauses_sql}"
+
         return sql
 
     def sql_information_schema_identifier(self) -> str:
         return "information_schema.tables"
 
+    ######################
+    # Query Execution
+    ######################
+
+    def get_row_counts_for_all_tables(
+        self, include_tables: list[str] | None, exclude_tables: list[str] | None, query_name: str | None
+    ) -> dict[str, int]:
+        """
+        Returns a dict that maps table names to row counts.
+        """
+        sql = self.sql_get_table_names_with_count(include_tables=include_tables, exclude_tables=exclude_tables)
+        if sql:
+            query = Query(
+                data_source_scan=self,
+                unqualified_query_name=query_name or "get_row_counts_for_all)tables",
+                sql=sql,
+            )
+            query.execute()
+            return {row[0]: row[1] for row in query.rows}
+
+        # Single query to get the metadata not available, get the counts one by one.
+        all_tables = self.find_table_names()
+        result = {}
+
+        for table in all_tables:
+            query = Query(
+                data_source_scan=self,
+                unqualified_query_name=f"get_counts_by_tables_for_row_count_anomalies_{table}",
```

Comment:
Why is this called row count anomalies? It seems that this method should be agnostic no?

Reply:
thanks for catching that, changed.